helm/aws-operator-chart/templates/deployment.yaml

```diff
@@ -1,50 +1,53 @@
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: aws-operator
-  namespace: {{ .Values.namespace }}
+  name: {{ tpl .Values.resource.default.name  . }}
+  namespace: {{ tpl .Values.resource.default.namespace  . }}
   labels:
-    app: aws-operator
+    app: {{ tpl .Values.resource.default.name  . }}
 spec:
   replicas: 1
   revisionHistoryLimit: 3
+  selector:
+    matchLabels:
+      app: {{ tpl .Values.resource.default.name  . }}
   strategy:
     type: RollingUpdate
   template:
     metadata:
       annotations:
         releasetime: {{ $.Release.Time }}
       labels:
-        app: aws-operator
+        app: {{ tpl .Values.resource.default.name  . }}
     spec:
       volumes:
-      - name: aws-operator-configmap
+      - name: {{ .Values.project.name }}-configmap
         configMap:
-          name: aws-operator-configmap
+          name: {{ tpl .Values.resource.default.name  . }}
```

Comment:
What's the difference between following:
- `aws-operator-configmap`
- `{{ .Values.project.name }}-configmap`
- `{{ tpl .Values.resource.default.name . }}`

They used to be the same, but now there's difference. Is this intentional?

Is the idea to template per-deployment (for different VOO versions etc.) name via `.Values.resource.default.name` but keep the YAML content the same by static `.Values.project.name`?

Reply:
More info here https://github.com/giantswarm/azure-operator/pull/721#discussion_r393605620